tools/SourceKit/include/SourceKit/Core/LangSupport.h

```diff
@@ -302,14 +302,67 @@ class OptionsDictionary {
 struct Statistic;
 typedef std::function<void(ArrayRef<Statistic *> stats)> StatisticsReceiver;
 
+// Used to wrap the result of a request. There are three possibilities:
+// - The request succeeded (`value` is valid)
+// - The request was cancelled
+// - The request failed (with an `error`)
+template <typename T>
+class RequestResult {
+private:
+  enum Type {
+    Value,
+    Error,
+    Cancelled
+  };
+  union {
+    const T *data;
+    StringRef error;
```

Comment:
How do we know this string is still alive?  Since this data structure looks general, I think this ought to own the storage (e.g. use a `std::string`).

Reply:
It's not just the error string, the value also is captured as a pointer and it may not be alive.

Reply:
Ah right.  We should document that loudly in the doc comment of the type then.  It's not safe to return/store this thing yada yada yada.

Reply:
Commented